crates/net/eth-wire/src/types/message.rs

```diff
@@ -287,6 +289,26 @@ pub trait EncodableExt {
     /// Users of this method should be aware that the truncated data may not represent the complete
     /// original dataset.
     fn encode_truncate(&self, approx: usize, limit: usize) -> Vec<u8>;
+
+    /// Encodes the data up to the specified size limit, returning the encoded data along with
+    /// a flag indicating whether the limit was exceeded. This method is useful for partial
+    /// encoding scenarios where it is necessary to encode as much as possible without going
+    /// over the limit.
+    ///
+    /// # Arguments
+    ///
+    /// * `approx` - The approximate size to allocate in memory for the encoding process. This is
+    ///   used to pre-allocate memory for efficiency.
+    /// * `limit` - The maximum allowed size of the encoded data.
+    ///
+    /// # Returns
+    ///
+    /// A tuple containing:
+    /// * The encoded data as `Vec<u8>`.
+    /// * The length of the data at the point of reaching the limit, or the 0 length if the limit
+    ///   was not exceeded.
+    /// * A boolean flag that is `true` if the size limit was exceeded and `false` otherwise.
```

Comment:
this is not needed. the caller's context can just check if the length of the data before limit was exceeded is zero

Reply:
Not sure what you mean here :)

Reply:
left this suggestion some lines lower but now that a commit was pushed it jumped up.


the boolean isn't needed.